torch/fx/_symbolic_trace.py

```diff
@@ -693,6 +694,12 @@ def trace(
         _is_fx_tracing_flag = True
         try:
             if isinstance(root, torch.nn.Module):
+
+                # do real recompilation for LazyGraphModule before retracing
+                from torch.fx.lazy_graph_module import LazyGraphModule
+                if isinstance(root, LazyGraphModule):
+                    root.real_recompile()
```

Comment:
Why do we sometimes need to manually trigger recompiles?

Reply:
It's mainly about retracing. Say we have got a LazyGraphModule and we want to retrace that with `symbolic_trace`. Then the `_lazy_forward` method will be traced. `symbolic_trace` does not support that and I got error: https://gist.github.com/shunting314/75549c2e82ae07ac1139c94a3583d259 . I think it's much easier to just force the recompilation before retracing with `symbolic_trace`.